online-fishing-store/src/app/pages/shopping-cart-page/shopping-cart-item/shopping-cart-item.component.ts

```diff
@@ -0,0 +1,27 @@
+import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { ShoppingCartProduct } from 'src/app/interfaces/shopping-cart-product';
+
+@Component({
+  selector: 'app-shopping-cart-item',
+  templateUrl: './shopping-cart-item.component.html',
+  styleUrls: ['./shopping-cart-item.component.scss']
+})
+export class ShoppingCartItemComponent  {
+  @Input() public shoppingCartProduct: ShoppingCartProduct;
+  @Output() public removeShoppingCartProduct: EventEmitter <ShoppingCartProduct> = new EventEmitter<ShoppingCartProduct>()
+  @Output() public addItem: EventEmitter <ShoppingCartProduct> = new EventEmitter<ShoppingCartProduct>()
+
+  public productAltImgAttribute: string = 'photo';
+  public quantity: number = 1;
+
+  public addItemProduct(): void { 
+    let productQuantity = {...this.shoppingCartProduct};
```

Comment:
It is brainstorm!!

How it can be possible productQuantity is equal this.shoppingCardButton.
If you have product quantity, it should has number type. this.shoppingCartProduct it is ShoppingCartProduct type